tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -337,54 +343,81 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 
 				global.gc();
 				await testObject.run();
 
 				await testObject.afterIteration?.();
 
 				global.gc();
-				benchmarkStats.samples.after.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.after.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.after.heapSpace.push(v8.getHeapSpaceStatistics());
 
-				benchmarkStats.runs++;
+				sample.after.memoryUsage.push(process.memoryUsage());
+				sample.after.heap.push(v8.getHeapStatistics());
+
+				sample.after.heapSpace.push(v8.getHeapSpaceStatistics());
+
+				runs++;
 
 				const heapUsedArray: number[] = [];
-				for (let i = 0; i < benchmarkStats.samples.before.memoryUsage.length; i++) {
+				for (let i = 0; i < sample.before.memoryUsage.length; i++) {
 					heapUsedArray.push(
-						benchmarkStats.samples.after.memoryUsage[i].heapUsed -
-							benchmarkStats.samples.before.memoryUsage[i].heapUsed,
+						sample.after.memoryUsage[i].heapUsed -
+							sample.before.memoryUsage[i].heapUsed,
 					);
 				}
 				heapUsedStats = getArrayStatistics(heapUsedArray, options.samplePercentageToUse);
 
 				// Break if max elapsed time passed, only if we've reached the min sample count
 				if (
-					benchmarkStats.runs >= options.minSampleCount &&
+					runs >= options.minSampleCount &&
 					timer.toSeconds(startTime, timer.now()) > options.maxBenchmarkDurationSeconds
 				) {
 					break;
 				}
 			} while (
-				benchmarkStats.runs < options.minSampleCount ||
+				runs < options.minSampleCount ||
 				heapUsedStats.marginOfErrorPercent > options.maxRelativeMarginOfError
 			);
-			benchmarkStats.stats = heapUsedStats;
+
+			benchmarkStats.customDataFormatters.Iteration = (value): string =>
+				(value as number).toString();
+
+			benchmarkStats.customData["Heap Used Avg"] = heapUsedStats.arithmeticMean;
+			benchmarkStats.customDataFormatters["Heap Used Avg"] = (value): string =>
+				prettyNumber(value as number, 2);
+
+			benchmarkStats.customData["Margin of Error"] = heapUsedStats.marginOfError;
+			benchmarkStats.customDataFormatters["Margin of Error"] = (value): string =>
+				`±${prettyNumber(value as number, 2)}`;
+
+			benchmarkStats.customData["Relative Margin of Error"] =
+				heapUsedStats.marginOfErrorPercent;
+			benchmarkStats.customDataFormatters["Relative Margin of Error"] = (value): string =>
+				`±${prettyNumber(value as number, 2)}`;
+
+			benchmarkStats.customData["Heap Used StdDev"] = heapUsedStats.standardDeviation;
+			benchmarkStats.customDataFormatters["Heap Used StdDev"] = (value): string =>
+				prettyNumber(value as number, 2);
+
+			// benchmarkStats.customData["Samples used"] = sample;
+			benchmarkStats.customData.Iterations = runs;
 		} catch (error) {
 			// TODO: This results in the mocha test passing when it should fail. Fix this.
-			benchmarkStats.aborted = true;
-			benchmarkStats.error = error as Error;
+			benchmarkStats = {
+				error: (error as Error).message,
+			};
```

Comment:
Check if this will work? Based on how the code worked before, I think it should. Before we would be keeping all the other properties in `benchmarkStats` if we hit this line, so I think it's better to maintain that behavior just in case. If this complains, maybe `(benchmarkStats as BenchmarkError).error = ...`.
```suggestion
			benchmarkStats.error = (error as Error).message;
```

Reply:
I gave a `BenchmarkData` initial value to `benchmarkStats`. I think it make sense to re-create a new `BenchamrkError`.

Reply:
Hmm but it's declared as `benchmarkStats: BenchmarkResult` on line 315, I think precisely so we can make it into an object that has the `error` property here and is thus a `BenchmarkError`. Although maybe you're right, in case of error we don't output anything from the `BenchmarkData` anyway. Yeah, we can create a new object here.